src/components/toast-notification-manager/daikin-toast-notification-manager.ts

```diff
@@ -0,0 +1,233 @@
+ *     <span slot="title">Toast title</span>
+ *     <span slot="description">Toast description</span>
+ *   </daikin-toast-notification>
+ * </daikin-toast-notification-manager>
+ * ```
+ */
+@customElement("daikin-toast-notification-manager")
+export class DaikinToastManager extends LitElement {
+  static override readonly styles = css`
+    ${unsafeCSS(tailwindStyles)}
+
+    :host {
+      display: block;
+    }
+  `;
+
+  /**
+   * Specify where on the screen the toast will be displayed.
+   */
+  @property({ type: String, reflect: true })
+  position: ToastPositionType = "bottom-right";
+
+  /**
+   * Specify how many seconds after the toast is displayed the close event should be fired (unit = ms).
+   * If `null`, there will be no automatic close event firing.
+   */
+  @property({ type: Number, reflect: true })
+  duration: number | null = null;
+
+  @queryAssignedElements({ selector: "daikin-toast-notification" })
+  private readonly _items!: readonly DaikinToastNotification[];
+
+  @state()
+  private _beforeItems: readonly DaikinToastNotification[] = [];
+
+  @state()
+  _newItem: DaikinToastNotification | null = null;
+
+  private _containerRef = createRef<HTMLElement>();
+
+  get _positionY(): "top" | "bottom" {
+    return this.position.startsWith("top") ? "top" : "bottom";
+  }
+
+  private _close(target: DaikinToastNotification) {
+    const items = this._items;
+    const targetIndex = items.findIndex((item) => item === target);
+    const afterItems = items.filter((_, index) =>
+      this._positionY === "top" ? index > targetIndex : index < targetIndex
+    );
+
+    for (const item of items) {
+      item.style.removeProperty(`--transition-duration`);
+    }
+
+    target.style.setProperty(`--move-offset-x`, TOAST_MOVE_OFFSET_X);
+    target.style.setProperty(`--opacity`, "0");
+    target.style.setProperty(`--pointer-events`, "none");
+
+    for (const item of afterItems) {
+      const height = item.clientHeight;
+      item.style.setProperty(
+        `--move-offset-y`,
+        `calc(0.5rem + ${
+          height *
+          (this._positionY === "top"
+            ? TOAST_MOVE_OFFSET_Y_TOP_SIGN
+            : TOAST_MOVE_OFFSET_Y_BOTTOM_SIGN)
+        }px)`
+      );
+    }
+
+    setTimeout(() => {
+      for (const item of this._items) {
+        item.style.setProperty(`--transition-duration`, "0");
+      }
+
+      this.dispatchEvent(
+        new CustomEvent("close", { detail: { name: target.name } })
+      );
+      for (const item of this._items) {
+        item.style.removeProperty(`--move-offset-y`);
+      }
+    }, TOAST_DURATION);
+  }
+
+  private _handleClose(event: Event) {
+    event.stopPropagation();
+
+    this._close(event.target as DaikinToastNotification);
+  }
+
+  private _handleSlotchange() {
+    const items = this._items;
+
+    this._newItem =
+      items.find(
+        (item) => !this._beforeItems.find((beforeItem) => item === beforeItem)
+      ) ?? null;
+    this._beforeItems = items;
```

Comment:
- Call method directly instead of via state.
- Handle cases where multiple elements are added at once.
- Use `Set` for performance.

```suggestion
    const newItems = items.filter((item) => !this._knownItemSet.has(item));
    this._knownItemSet = new Set(items);

    for (const newItem of newItems) {
      this._handleNewItem(newItem);
    }
```